lib/addressable/uri.rb

```diff
@@ -1094,7 +1098,7 @@ def host=(new_host)
 
       unreserved = CharacterClasses::UNRESERVED
       sub_delims = CharacterClasses::SUB_DELIMS
-      if @host != nil && (@host =~ /[<>{}\/\?\#\@"]/ ||
+      if @host != nil && (@host =~ /[<>{}\/\?\#\@"\r\n]/ ||
```

Comment:
Maybe something like `[[:space:]]` would be more general purpose and allow us to catch more whitespace related values?